### PR TITLE
Balder/document weight iterator must support term wise

### PR DESCRIPTION
**Pull request by baldersheim**
@havardpe PR, No semantic changes ( at least not intentional)

### CODE REVIEW

File: searchlib/src/tests/attribute/searchable/attribute_searchable_adapter_test.cpp [hunk 1]
```diff
@@ -472,6 +464,26 @@ TEST("require that direct attribute iterators work") {
...
+        bool strict = true;
+        bool isFilter = true;
+        MyAttributeManager attribute_manager = make_weighted_string_attribute_manager(fast_search, isFilter);
+        SimpleStringTerm empty_node("notfoo", "", 0, Weight(1));
+        Result empty_result = do_search(attribute_manager, empty_node, strict);
+        EXPECT_EQUAL(0u, empty_result.hits.size());
+        SimpleStringTerm node("foo", "", 0, Weight(1));
+        Result result = do_search(attribute_manager, node, strict);
+        EXPECT_EQUAL(3u, result.est_hits);
+        EXPECT_TRUE(result.iterator_dump.find("DocumentWeightSearchIterator") == vespalib::string::npos);
```
Comment (havardpe) [inline searchlib/src/tests/attribute/searchable/attribute_searchable_adapter_test.cpp]: how does not using the DocumentWeightSearchIterator test that it supports termwise evaluation?

**Review by havardpe — COMMENTED** [review]
some minor comments, plus me not understanding the test...

**Comment by havardpe** [comment]
attribute_searchable_adapter_test.cpp:479: error: check failure #1: 'result.iterator_dump.find("FilterAttributePostingListIteratorT") == vespalib::string::npos' in thread '0(1)' (attribute_searchable_adapter_test.cpp)
    STATE[0]: 'require that single weighted set turns filter on filter fields' (attribute_searchable_adapter_test.cpp:467)

ok, so you are testing that a weighted set tagged as filter with a single element becomes a filter iterator and not a document weight iterator... which makes the pull request title a bit misleading...

File: searchlib/src/tests/attribute/searchable/attribute_searchable_adapter_test.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/tests/attribute/searchable/attribute_searchable_adapter_test.cpp]: It does not test that it supports termwise, it just tests that it instantiates a filter iterator.

Comment (havardpe) [reply searchlib/src/tests/attribute/searchable/attribute_searchable_adapter_test.cpp]: but it is the 'weighted set blueprint' and not the 'document weight iterator' that is tested? (hence my confusion)

**Comment by baldersheim** [comment]
Correct, the pull request was made before the root cause was fully investigated. It just reflected the what was first observed and concluded to be incorrect.

However it turned out to be nothing wrong with the code. It was missing restart detection at deploy time....

File: searchlib/src/vespa/searchlib/queryeval/leaf_blueprints.cpp [hunk 2]
```diff
@@ -67,30 +62,22 @@ SimpleBlueprint::tag(const vespalib::string &t)
...
-FakeBlueprint::~FakeBlueprint()
-{
-}
+FakeBlueprint::~FakeBlueprint() = default;
 
-//-----------------------------------------------------------------------------
 
-} // namespace queryeval
-} // namespace search
+}
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/queryeval/leaf_blueprints.cpp]: Fixed

**Comment by baldersheim** [comment]
All comments addressed, merging

**Pull request merged**